src/archetype/ArchetypeModule.ts

```diff
@@ -140,6 +146,59 @@ async function executeInTerminalHandler(metadata: IProjectCreationMetadata): Pro
     vscode.tasks.executeTask(createProjectTask);
 }
 
+async function createBasicMavenProject(metadata: IProjectCreationMetadata): Promise<void> {
+    const {
+        groupId,
+        artifactId,
+        targetFolder
+    } = metadata;
+    if (!groupId || !artifactId || !targetFolder) {
+        return;
+    }
+
+    const task = async (p: vscode.Progress<{ message?: string; increment?: number }>) => {
+        // copy from template
+        p.report({ message: "Generating project from template...", increment: 20 });
+        const templateUri = vscode.Uri.file(getPathToExtensionRoot("resources", "projectTemplate"));
+        const targetUri = vscode.Uri.joinPath(vscode.Uri.file(targetFolder), artifactId);
+        await workspace.fs.copy(templateUri, targetUri, { overwrite: true });
+
+        // update groupId/artifactId in pom.xml
+        p.report({ message: "Updating pom.xml file...", increment: 20 });
+        const pomUri = vscode.Uri.joinPath(targetUri, "pom.xml");
+        let pomContent = (await workspace.fs.readFile(pomUri)).toString();
+        pomContent = pomContent.replace("${groupId}", groupId);
+        pomContent = pomContent.replace("${artifactId}", artifactId);
+        await workspace.fs.writeFile(pomUri, Buffer.from(pomContent));
+
+        // create source files
+        p.report({ message: "Creating source files...", increment: 20 });
+        await vscode.workspace.fs.createDirectory(vscode.Uri.joinPath(targetUri, "src", "main", "java"));
```

Comment:
how about saving the whole template project to `./resources/projectTemplate` so that we only need to copy the folder here?

Reply:
For two reasons:
1. empty folder (e.g. src/test/java) cannot be checked into git tree
2. the folder structure of `src/main/java` is not fixed, depending on `groupId`

Reply:
ok, let's keep the current implementation